ui/desktop/electron-app/src/cli/index.js

```diff
@@ -5,11 +5,27 @@
 
 const { spawnSync } = require('../helpers/spawn-promise.js');
 const { path } = require('./path.js');
+const pathPosix = require('path');
+const pathWin = require('path/win32');
+const { isWindows } = require('../helpers/platform.js');
 
 module.exports = {
   // Check boundary cli existence
   exists: () => Boolean(path()),
 
+  // Return true if the CLI in usage is the built in. False if relies on system CLI.
+  isBuiltInCli: () => {
+    const pathToParse = path();
```

Comment:
Do we need to use the path from `path()`?

Couldn't we have pulled out the logic for `isBuiltInCli` to another method and used that?

```javascript
    const builtInCliPath = isDev
      ? path.resolve(__dirname, '..', '..', 'cli', binaryName)
      : path.resolve(process.resourcesPath, 'cli', binaryName);
    const isBuiltInCli = existsSync(builtInCliPath);
```

Reply:
@ZedLi 

If understand correctly, you suggest to move the `isBuiltInCli` to the `path.js` file? 
Initially I though of extracting to a function the current logic under `path()`, and re-use it for method path and isBuiltInCli, since is the same logic.
But discarted since that logic lives in `path.js` and not in `index.js`, where the current `exists()` method lives.

Let me know if you mean to move the logic to `path.js` and I will proceed, or feel free to ping me and clarify through zoom.

Thank you

Reply:
Yeah basically, I didn't realize the names are the same but I was specifically talking about the logic of `isBuiltInCli` from `path()`.

I'm not sure we need the actual path itself as much as the logic that checks if we have our own CLI, which means we could then skip doing the parsing of the path after

Reply:
100%. Yeah Indeed, I will proceed to move `isBuiltInCli` to `path.js` so we can share the logic.

This, despite is an easy check, I had doubts where will be best approach and best place to make it happend. 

Iterations will provide the better answer 😃